### PR TITLE
Implement services required to implement fatal.

### PR DESCRIPTION
**Pull request by SciresM**
This adds...a lot of wrappers, sufficient to implement the fatal system module.

### CODE REVIEW

File: nx/include/switch/services/gpio.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+Result gpioInitialize(void);
+void gpioExit(void);
+
+Result gpioOpenSession(GpioPadSession *out, GpioPadName name);
+
+Result gpioPadSetDirection(GpioPadSession *p, GpioDirection dir);
+Result gpioPadGetDirection(GpioPadSession *p, GpioDirection *out);
+Result gpioPadSetValue(GpioPadSession *p, GpioValue val);
+Result gpioPadGetValue(GpioPadSession *p, GpioValue *out);
+void gpioPadClose(GpioPadSession *p);
```
Comment (fincs) [inline nx/include/switch/services/gpio.h]: Missing newline at eof

File: nx/include/switch/services/i2c.h [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    I2cTransactionOption_All = I2cTransactionOption_Start | I2cTransactionOption_Stop,
+} I2cTransactionOption;
+
+Result i2cInitialize(void);
+void i2cExit(void);
+
+Result i2cOpenSession(I2cSession *out, I2cDevice dev);
+
+Result i2csessionSendAuto(I2cSession *s, void *buf, size_t size, I2cTransactionOption option);
+void i2csessionClose(I2cSession *s);
```
Comment (fincs) [inline nx/include/switch/services/i2c.h]: Missing newline at eof

File: nx/source/services/bpc.c [hunk 3]
```diff
@@ -33,61 +33,98 @@ Result bpcShutdownSystem(void)
...
+
+        rc = resp->result;
+
+        if (R_SUCCEEDED(rc)) {
+            *out = (BpcSleepButtonState)resp->state;
+        }
+    }
+
+    return rc;
+}
```
Comment (fincs) [inline nx/source/services/bpc.c]: Missing newline at eof

File: nx/source/services/set.c [hunk 4]
```diff
@@ -613,3 +613,76 @@ Result setsysGetFirmwareVersion(SetSysFirmwareVersion *out) {
...
+        rc = resp->result;
+
+        if (R_SUCCEEDED(rc)) {
+            *flags_0 = resp->flags_0;
+            *flags_1 = resp->flags_1;
+        }
+    }
+
+    return rc;
+}
```
Comment (fincs) [inline nx/source/services/set.c]: Missing newline at eof

File: nx/include/switch/services/fs.h [hunk 5]
```diff
@@ -254,5 +254,5 @@ void fsEventNotifierClose(FsEventNotifier* e);
 Result fsDeviceOperatorIsSdCardInserted(FsDeviceOperator* d, bool* out);
 Result fsDeviceOperatorIsGameCardInserted(FsDeviceOperator* d, bool* out);
 Result fsDeviceOperatorGetGameCardHandle(FsDeviceOperator* d, FsGameCardHandle* out);
-Result fsDeviceOperatorGetGameCardAttribute(FsDeviceOperator* d, FsGameCardHandle handle, u8 *out);
+Result fsDeviceOperatorGetGameCardAttribute(FsDeviceOperator* d, FsGameCardHandle* handle, u8 *out);
```
Comment (fincs) [inline nx/include/switch/services/fs.h]: Should probably be a const pointer

File: nx/source/services/set.c [hunk 6]
```diff
@@ -613,3 +613,76 @@ Result setsysGetFirmwareVersion(SetSysFirmwareVersion *out) {
...
+
+        struct {
+            u64 magic;
+            u64 result;
+        } *resp = r.Raw;
+
+        rc = resp->result;
+
+        if (R_SUCCEEDED(rc)) {
+            eventLoadRemote(out, r.Handles[0], false);
```
Comment (yellows8) [inline nx/source/services/set.c]: The .h should mention that autoclear is false.

**Pull request merged**